visor/visor.go

```diff
@@ -176,12 +178,17 @@ func (v *Visor) Run(ctx context.Context) error {
 
 					numOfUpgradeStatusErrs++
 
-					v.log.Info("Still waiting for vega to start...", logging.Int("attemptLeft", maxUpgradeStatusErrs-numOfUpgradeStatusErrs))
+					attemptsLeft := maxUpgradeStatusErrs - numOfUpgradeStatusErrs
+					if numOfRestarts == 0 {
+						attemptsLeft = maxNumberOfFirstConnectionRetries - numOfUpgradeStatusErrs
+					}
+					v.log.Info("Still waiting for vega to start...", logging.Int("attemptLeft", attemptsLeft))
 
 					break
 				}
 
 				if !upStatus.ReadyToUpgrade {
+					numOfUpgradeStatusErrs = 0
```

Comment:
Not sure why do you reset it here?

Reply:
No reason other than it feels like that the counter should be for the number of times of times we can't get the upgrade status in a row?

Reply:
Yes I guess that's fair.